xlml/apis/task.py

```diff
@@ -503,3 +504,118 @@ def clean_up(self, resource: airflow.XComArg, project_id: str, zone: str) -> DAG
       AirflowTaskTimeout: An error occurs when execution_timeout is breached.
     """
     return gpu.delete_resource.override(group_id="clean_up")(resource, project_id, zone)
+
+
+@dataclasses.dataclass
+class GpuGkeTask(BaseTask):
+  """This is a class to set up tasks for GPU.
+
+  Attributes:
+    image_project: the project that an image belongs to.
+    image_family: the family group that an image belongs to.
+  """
+
+  task_test_config: test_config.JSonnetGpuTest
+  task_gcp_config: gcp_config.GCPConfig
+  cluster_name: str
+  job_create_timeout: datetime.timedelta = datetime.timedelta(minutes=10)
+  # TODO: metrics
+  # task_metric_config: Optional[metric_config.MetricConfig] = None
+
+  def run(self) -> DAGNode:
+    """Run a test job.
+
+    Returns:
+      A task group with the following tasks chained: provision, run_model,
+      post_process, clean_up.
+    """
+    # piz: We skip the queued resource for GPU for now since there is no queued
+    # resource command for GPU.
+    with TaskGroup(
+        group_id=self.task_test_config.benchmark_id, prefix_group_id=True
+    ) as group:
+      job_body = self._get_job_manifest()
+      gke.run_job(
```

Comment:
I understand when more options/workflows come, we will diverge the name a little bit. But I think it will be good if we keep the name aligned as much as possible, i.e. still use run_model and post_process if applies. (

This is one [example](https://screenshot.googleplex.com/BtQsZgw4gsXR3Pz) for GKE on TPU, I am thinking we could rename something below. 

In this case:

```
run_model
* deploy_workload
* stream_logs
post_process
```

In xpk case (I can do the change):

```
run_model
* generate_workload_id
* deploy_workload
* wait_for_workload_completion
post_process
```

I like `workload` better as job can be a end-to-end test job in general.

Reply:
Renaming the inner tasks SGTM. I think the top-level Python name `gke.run_job` makes sense here because it is taking a Kubernetes `Job` here. "Job" ends up being overloaded in general, but I this actually means `Job` in a specific sense, which is distinct from other types of workloads you can run on GKE.

Reply:
IIRC it is possible to overload the name of a decorated task group, so we could keep the Python API of `gke.run_job` and generate a task group named `run_model`.

Reply:
Sounds good to me with the task group named run_model. Will we have structure like this on UI?

```
run_model 
* deploy_workload 
* stream_logs 
post_process
```

Reply:
Right. Added the override to name the task group `run_model`